lib/bgs/exceptions/bgs_errors.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'bgs/exceptions/service_exception'
+module BGS
+  module Exceptions
+    module BGSErrors
+      include SentryLogging
+      MAX_ATTEMPTS = 3
+
+      def with_multiple_attempts_enabled
+        attempt ||= 0
+        yield
+      rescue => e
+        attempt += 1
+        if attempt < MAX_ATTEMPTS
```

Comment:
Do you want `<= MAX_ATTEMPTS?`?

Reply:
It's kind of no big deal. I caught a glimpse of the Java code we're attempting to replace with this feature and they try 3 times.

I think since it increments before the conditional it's working 3 times. I verified with tests. 👍